wormhole/database.py

```diff
@@ -104,3 +104,49 @@ def dump(self) -> dict:
             "guild_id": self.guild_id,
             "channel_id": self.channel_id,
         }
+
+
+class WormholePatterns(database.base):
+    __tablename__ = "wormhole_wormhole_wormholepatterns"
+
+    id = Column(Integer, primary_key=True, autoincrement=True)
+    regex_pattern = Column(String(255), nullable=False)
+    replacement = Column(Text, nullable=False)
+
+    @classmethod
+    def set_pattern(cls, regex_pattern: str, replacement: str):
+        """
+        Adds a new pattern or updates it if it already exists.
+        """
+        pattern = session.query(cls).filter_by(regex_pattern=regex_pattern).first()
+        if pattern:
+            pattern.replacement = replacement
+        else:
+            pattern = cls(regex_pattern=regex_pattern, replacement=replacement)
+            session.add(pattern)
+        session.commit()
+        return pattern
+
+    @classmethod
+    def remove_pattern(cls, regex_pattern: str) -> bool:
```

Comment:
In Strawberry, remove should be called on already found DB object (first find, then delete). Also, there's the same problem as in add function - pattern is not a primary key.